servicetalk-http-api/src/main/java/io/servicetalk/http/api/HeaderUtils.java

```diff
@@ -656,24 +659,25 @@ private HttpCookiePair findNext(CharSequence cookieHeaderValue) {
     }
 
     /**
-     * Advertise the list of supported encodings to the request/response headers.
+     * Advertise the list of supported encodings to the request headers.
      * The list will be advertised as part of the Accept-Encoding header
      *
      * If {@code headers} already contain {@code 'Accept-Encoding'} header, then this method has no effect.
      *
-     * @param headers the headers to modify
+     * @param metaData the metadata headers to modify
      * @param encodings the list of encodings to be used in the string representation.
      * @see <a href="https://tools.ietf.org/html/rfc7231#page-41">Accept-Encodings</a>
      */
-    public static void advertiseAcceptedEncodingsIfAvailable(final HttpHeaders headers,
-                                                  final Collection<ContentCoding> encodings) {
+    public static void advertiseAcceptedEncodingsIfAvailable(final HttpMetaData metaData,
+                                                  final Collection<StreamingContentCodec> encodings) {
+        final HttpHeaders headers = metaData.headers();
         if (headers.contains(ACCEPT_ENCODING) || encodings.isEmpty()) {
             return;
         }
 
         StringBuilder builder = new StringBuilder();
-        for (ContentCoding enc : encodings) {
-            if (enc == none()) {
+        for (StreamingContentCodec enc : encodings) {
+            if (enc == identity()) {
```

Comment:
This is always considered acceptable, unless specifically disabled with a 'q' value. Its allowed to be skipped.
I skipped it, to avoid having it in the headers if its the only encoding, it provides no useful info.